lib/jobs/ipmi-job.js

```diff
@@ -184,20 +184,117 @@ function ipmiJobFactory(
     };
 
     /**
-     * Collect SEL entries list from IPMI
+     * gets the SEL information using the "sel get" ipmi command
      * @memberOf IpmiJob
      *
      * @param data
-     * @param count
+     * @param stringifiedEntries
      */
-    IpmiJob.prototype.collectIpmiSel = function(data, count) {
-        count = count || 25;
-        return ipmitool.sel(data.host, data.user, data.password, count)
-        .then(function (sel) {
-            return parser.parseSelData(sel);
-        });
+    IpmiJob.prototype.getSelEntries = function(data, stringifiedEntries) {
+        return ipmitool.sel(data.host, data.user, data.password,  stringifiedEntries)
+    };
+
+    /**
+     * gets the raw information of a single sel entry
+     * @memberOf IpmiJob
+     *
+     * @param data
+     * @param recordId
+     */
+    IpmiJob.prototype.getRawSelEntry = function(data, recordId) {
+        return ipmitool.readRawSel(data.host, data.user, data.password, recordId)
+    };
+
+    /**
+     * Collect SEL entries list from IPMI and add two properties ("Sensor Type Code" and "Event Type Code") to each sel entry
+     * @memberOf IpmiJob
+     *
+     * @param data
+     */
+    IpmiJob.prototype.collectIpmiSel = function(data) {
+        var self = this;
+        var stringifiedEntries = "";
+        var lastUpdatedSelDate;
+        var currentSelDate;
+        var lastReportedSelEntryID;
+        var totalEntries;
+        var lastUpdatedSelDate;
+        var sel;
+
+        return Promise.all([
+           self.collectIpmiSelInformation(data),
+           waterline.workitems.findOne({ id: data.workItemId })
+        ])
+        .spread (function(selInfo, workObj){
+            totalEntries = parseInt(selInfo.Entries);
+            var tmpDate = selInfo["Last Add Time"].split(/\/|:| /);
+            tmpDate =tmpDate[0] +" "+ tmpDate[1] +", "+ tmpDate[2] +" "+ tmpDate[3] +":"+ tmpDate[4]+":"+ tmpDate[5] ;
```

Comment:
If you're interested, you can probably write this a little easier like:

```
tmpDate = "%s %s, %s %s:%s:%s".format.apply(null, selInfo["Last Add Time"].split(/\/|:| /));
```

Because we've added `format` to the String prototype here: https://github.com/RackHD/on-core/blob/master/lib/extensions.js#L39